src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,336 @@
+            waitForStateOfCluster(clusterId, "STOPPED", TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+            return getHazelcastCloudCluster(clusterId);
+        } catch (Exception e) {
+            throw new CloudException(String.format("Stop cluster with id %s is failed, Rc stack trace is: %s", clusterId, e.getStackTrace()));
+        }
+    }
+
+    public CloudCluster resumeHazelcastCloudCluster(String clusterId) throws CloudException {
+        try {
+            String query = String.format("{\"query\": \"mutation { resumeCluster(clusterId:\\\"%s\\\") { clusterId }}\"}", clusterId);
+            String responseBody = prepareAndSendRequest(query).body().string();
+            LOG.info(maskValueOfToken(responseBody));
+            waitForStateOfCluster(clusterId, "RUNNING", TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+            return getHazelcastCloudCluster(clusterId);
+        } catch (Exception e) {
+            throw new CloudException(String.format("Resume cluster with id %s is failed, Rc stack trace is: ", clusterId, e.getStackTrace()));
+        }
+    }
+
+    public void deleteHazelcastCloudCluster(String clusterId) throws CloudException {
+        try {
+            String query = String.format("{\"query\": \"mutation { deleteCluster(clusterId:\\\"%s\\\") { clusterId }}\"}", clusterId);
+            String responseBody = prepareAndSendRequest(query).body().string();
+            LOG.info(responseBody);
+            waitForDeletedCluster(clusterId, TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+            LOG.info(String.format("Cluster with id %s is deleted", clusterId));
+        } catch (Exception e) {
+            throw new CloudException(String.format("Delete hazelcast cloud cluster with id %s is failed, Rc stack trace is: %s", clusterId, e.getMessage()));
+        }
+    }
+
+    private Response prepareAndSendRequest(String query) throws InterruptedException, CloudException {
+        int retryCountForExceptionOfEndpoint = 0;
+        // Rarely server returns empty header, that is why a retry mechanism is added.
+        while(retryCountForExceptionOfEndpoint < 3)
+        {
+            try {
+                RequestBody body = RequestBody.create(JSON, query);
+                Request request = new Request.Builder()
+                        .url(HttpUrl.get(uri))
+                        .post(body)
+                        .header("Authorization", String.format("Bearer %s", bearerToken))
+                        .header("Content-Type", "application/json")
+                        .build();
+
+                call = client.newCall(request);
+                return call.execute();
+            }
+            catch(Exception e)
+            {
+                Log.warn(e.toString());
+            }
+            retryCountForExceptionOfEndpoint++;
+            TimeUnit.SECONDS.sleep(2);
+        }
+        throw new CloudException("Request cannot send successfully after 3 tries");
```

Comment:
Please keep the last exception e in a variable and put its stack trace to the cloud exception message.